src/main/java/MicroRaiden.java

```diff
@@ -85,8 +212,529 @@ public void listBalances(String account, String remotePeer) {
+                createChannelERC20FunctionBytes, 42);// chainid
+        createTrans.sign(keyPair);
+        String signedChannelCreationTrans = "0x" + new String(org.apache.commons.codec.binary.Hex.encodeHex(createTrans.getEncoded()));
+        String createChannelSendRawTransactionString = "{\"method\":\"eth_sendRawTransaction\",\"params\":[\""
+                + signedChannelCreationTrans + "\"],\"id\":42,\"jsonrpc\":\"2.0\"}";
+        
+    	String myTransactionID2="";
+    	try {
+    		myTransactionID2=getHttpResponse(createChannelSendRawTransactionString);
+        }catch (IOException e) {
+        	System.out.println("Fail to execute HTTP request.");
+    		return;
+        }
+    	
+        if(!"".equals(myTransactionID2)) {
+	        String blockNumberHex=waitingForTransaction(myTransactionID2);
+	        if(debugInfo) {
+	        	System.out.println("\bChannel has been opened in block "+blockNumberHex);
+	        }
+	        Digest keccak256 = Digests.keccak256();
+
+	        
+	        String firstArgVal=address.substring(2).toLowerCase();
+	        String secondArgVal=receiverAccountID.substring(2).toLowerCase();
+	        String thirdArgVal=String.format("%08x", Integer.parseInt(blockNumberHex.substring(2), 16));
+	        try{
+	        	byte[] data = concatenateByteArrays(Hex.decodeHex(firstArgVal.toCharArray()),Hex.decodeHex(secondArgVal.toCharArray()),Hex.decodeHex(thirdArgVal.toCharArray()));
+	        	if(debugInfo) {
+	        		System.out.println("The keccak256 argument of bytes in string "+Hex.encodeHexString(data));	
+	        	}
+	        	byte[] keyInBytes=keccak256.reset().update(data).digest();
+	        	String channelKeyHex = "0x"+new String(Hex.encodeHexString(keyInBytes));
+	        	System.out.println("\bChannel key = "+channelKeyHex);
+	        	System.out.println("Channel on Koven can be found on page:\nhttps://kovan.etherscan.io/address/0x4913f12d38c04094cf1e382d0ffeef3036ecca32#readContract");
+	        }catch (DecoderException e) {
+	        	System.out.println("Hex string cannot be converted to byte array!");
+	        }        
+	        
+        }
+    	return;
+
+    }
+    
+    /**
+     * Helper function to join two byte arrays together.
+     * @param a the first byte array
+     * @param b the second byte array
+     * @return the merged array
+     */
+    private static byte[] concatenateByteArrays(byte[]... arrays)
+    {
+        int count = 0;
+        for (byte[] array: arrays)
+        {
+            count += array.length;
+        }
+
+        // Create new array and copy all array contents
+        byte[] mergedArray = new byte[count];
+        int start = 0;
+        for (byte[] array: arrays) {
+            System.arraycopy(array, 0, mergedArray, start, array.length);
+            start += array.length;
+        }
+        return mergedArray;
+    }
+    
+    private String getHttpResponse(String requestString) throws IOException{
```

Comment:
Similarly we may wish to move this function into either a helper class, or maybe one dedicated just to http / network comms. or something.